Network/Converter/PacketConverter.cs

```diff
@@ -27,321 +28,680 @@
-                else propertyData.SetValue(ReadPrimitiveFromStream(arrayType, binaryReader), i);
+                propertiesToSerialise[i].SetValue(
+                    obj,
+                    DeserialiseObjectFromReader(
+                        obj, propertiesToSerialise[i], binaryReader));
             }
 
-            return propertyData;
+            return obj;
         }
 
         /// <summary>
-        /// Reads the length of the list from the stream and creates the list instance.
-        /// Also fills the list by using recursion.  
+        /// Deserialises the given <see cref="PropertyInfo"/> from the given
+        /// <see cref="BinaryReader"/>s underlying <see cref="MemoryStream"/>.
         /// </summary>
-        /// <param name="obj">The object.</param>
-        /// <param name="propertyInfo">The property information.</param>
-        /// <param name="binaryReader">The binary reader.</param>
-        /// <returns>System.Collections.IList.</returns>
-        private IList ReadListFromStream(object obj, PropertyInfo propertyInfo, BinaryReader binaryReader)
+        /// <param name="obj">
+        /// The <see cref="object"/> whose <see cref="PropertyInfo"/> value to
+        /// deserialise.
+        /// </param>
+        /// <param name="propertyInfo">
+        /// The <see cref="PropertyInfo"/> to deserialise from the given
+        /// <see cref="BinaryReader"/>s underlying <see cref="MemoryStream"/>.
+        /// </param>
+        /// <param name="binaryReader">
+        /// The <see cref="BinaryReader"/> frpm whose underlying <see cref="MemoryStream"/>
+        /// to deserialise the given <see cref="PropertyInfo"/>.
+        /// </param>
+        /// <returns>
+        /// The <see cref="object"/> deserialised from the <see cref="MemoryStream"/>.
+        /// This can be null if the <see cref="NetworkObjectState"/> is
+        /// <see cref="NetworkObjectState.Null"/>.
+        /// </returns>
+        private object DeserialiseObjectFromReader(
+            object obj, PropertyInfo propertyInfo, BinaryReader binaryReader)
         {
-            int listSize = binaryReader.ReadInt32();
-            Type listType = propertyInfo.PropertyType.GetGenericArguments()[0];
-            IList list = (IList)Activator.CreateInstance(typeof(List<>).MakeGenericType(listType));
-            for (int i = 0; i < listSize; i++)
+            Type propertyType = propertyInfo.PropertyType;
+
+            // we have an enumeration
+            if (propertyType.IsEnum)
             {
-                if (listType.IsClass && !IsPrimitive(listType)) list.Add(ReadObjectFromStream(Activator.CreateInstance(listType), binaryReader));
-                else list.Add(ReadPrimitiveFromStream(listType, binaryReader));
+                return Enum.Parse(propertyType, binaryReader.ReadString());
             }
 
-            return list;
+            // we have an array
+            if (propertyType.IsArray)
```

Comment:
No "{" for one line if statements

Reply:
This i dont agree with, because i like verbosity. Also, the previous version was a pain to read, as it was clumped together. In such cases the curly brackets help maintain good readablility by clearly deliniation the if statement. Also, since the indentation alone is hard to make out (my eyesight must be going bad lol) i feel that keeping the curly brackets is better overall. Am happy to discuss this more though.